thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/monitor/MonitorTaskRunner.java

```diff
@@ -120,11 +134,62 @@ private void executeMonitorUpdate(MonitorTaskInfo monitorTaskInfo) {
 
       // update detection health
       updateDetectionHealth();
+
+      // disable alerts that failed consecutively for a long time
+      disableLongFailedAlerts();
+
     } catch (Exception e) {
       LOG.error("Exception in monitor update task", e);
     }
   }
 
+  /**
+   * Disable the alert if it was updated before {@MAX_TASK_FAIL_DAYS} but there is no success run since then.
+   */
+  private void disableLongFailedAlerts() {
+    DetectionConfigManager detectionDAO = DAO_REGISTRY.getDetectionConfigManager();
+    List<DetectionConfigDTO> detectionConfigs = detectionDAO.findAllActive();
+    long currentTimeMills = System.currentTimeMillis();
+    long maxTaskFailMills = TimeUnit.DAYS.toMillis(MAX_FAILED_DISABLE_DAYS);
+    for (DetectionConfigDTO config : detectionConfigs) {
+      try {
+        Timestamp updateTime = config.getUpdateTime();
+        if (updateTime != null && config.getHealth() != null && config.getHealth().getDetectionTaskStatus() != null) {
+          long lastTaskExecutionTime = config.getHealth().getDetectionTaskStatus().getLastTaskExecutionTime();
+          if (updateTime.getTime() <= currentTimeMills - maxTaskFailMills && (lastTaskExecutionTime == -1L
```

Comment:
Can we skip the `lastTaskExecutionTime == -1L` check?
Isn't that already covered by `lastTaskExecutionTime <= currentTimeMills - maxTaskFailMills`?

Reply:
lastTaskExecutionTime == -1L is used for backward compatibility.
Currently we have many long failing alerts have -1L.